pandas/_testing/asserters.py

```diff
@@ -949,7 +949,11 @@ def assert_series_equal(
             obj=f"{obj}.index",
         )
 
-    if check_freq and isinstance(left.index, (DatetimeIndex, TimedeltaIndex)):
+    if (
+        check_freq
+        and isinstance(left.index, (DatetimeIndex, TimedeltaIndex))
```

Comment:
hmm, what hits this?

Reply:
On L959 we use `ridx.freq`. Initially we didn't check if `right.index` was also of type `DatetimeIndex` or `TimedeltaIndex` and therefore mypy wasnt sure if the index contained the `freq` attribute. This threw the following error

```python
error: "Index" has no attribute "freq"  [attr-defined]
```